shared/src/chain_state.rs

```diff
@@ -39,14 +39,35 @@ impl<CI: ChainIndex> ChainState<CI> {
     #[allow(clippy::too_many_arguments)]
     pub fn new(
         store: &Arc<CI>,
-        tip_header: Header,
-        total_difficulty: U256,
-        cell_set: CellSet,
-        proposal_ids: TxProposalTable,
-        tx_pool: TxPool,
-        txs_verify_cache: LruCache<H256, Cycle>,
         consensus: Arc<Consensus>,
+        txs_verify_cache_size: usize,
```

Comment:
`txs_verify_cache_size` is one field of `TxPoolConfig`, we can omit it here.

Reply:
Logically `txs_verify_cache_size` is not related with `TxPoolConfig`, you can see we use `txs_verify_cache` in `chain` or other modules, it is more like a shared object.

I'd rather leave it here for now.

Reply:
Please check this line:
https://github.com/nervosnetwork/ckb/blob/61855c678d7806e5dc733e3806757d2f47a0a47b/shared/src/shared.rs#L305

`txs_verify_cache_size` is copied from `TxPoolConfig`

Reply:
Fair enough